subsys/bluetooth/scan.c

```diff
@@ -1459,7 +1469,9 @@ static void filter_state_check(struct bt_scan_control *control,
 		notify_filter_matched(&control->device_info,
 				      &control->filter_status,
 				      control->connectable);
+#if CONFIG_BT_CENTRAL
 		scan_connect_with_target(control, addr);
+#endif /* CONFIG_BT_CENTRAL */
```

Comment:
Consider using the IS_ENABLED macro here and leave the `scan_connect_with_target` function unconditionally defined. This function should be removed by the linker when CONFIG_BT_CENTRAL is disabled.

Reply:
I considered using the `IS_ENABLED` macro, I use conditional compilation because:
Define the function `scan_connect_with_targe` unconditionally, would require adding additional compilation conditions inside the function itself. As well as in the functions `notify_connecting` and `notify_connecting_error`.
These functions use structure fields that are conditionally defined.
Adding more compilation conditions, will reduce the code readability and make it more difficult to detect bugs.